app/assets/javascripts/controllers/container_topology/container_topology_controller.js

```diff
@@ -187,18 +189,53 @@ angular.module('topologyApp', ['kubernetesUI', 'ui.bootstrap'])
     window.location.assign(url);
   }
 
+  function getIcon(d) {
```

Comment:
The usual complaint.. can you please add the class names as comments to each case? (for searchability)